src/lib/jobs/deep-research.ts

```diff
@@ -0,0 +1,236 @@
+      return;
+    }
+
+    // Execute search queries and aggregate results
+    const searchResults = await executeDeepSearchQueries(db, userId, queries);
+    
+    // Process results and cache them
+    await cacheDeepResearchResults(userId, conversationId, searchResults);
+    
+    console.log(`Deep research completed for conversation ${conversationId}`);
+  } catch (error) {
+    console.error(`Deep research failed for conversation ${conversationId}:`, error);
+  }
+}
+
+/**
+ * Generate search queries based on recent conversation messages
+ */
+async function generateSearchQueries(
+  userId: string,
+  messages: DeepResearchJobInput["messages"]
+): Promise<string[]> {
+  const schema = z
+    .object({ queries: z.array(z.string()).min(1).max(5) })
+    .describe("DeepResearchQueries");
+
+  // Format messages for context
+  const messageContext = messages
+    .map((m) => `<message role="${m.role}">${m.content}</message>`)
+    .join("\n");
+
+  // Use structured analysis to generate meaningful search queries
+  try {
+    const res = await performStructuredAnalysis({
+      userId,
+      systemPrompt: "You are a helpful research assistant tasked with generating search queries based on a conversation.",
+      prompt: `<system:info>You are processing a conversation to identify important topics or questions that should be researched in depth.</system:info>
+
+<conversation>
+${messageContext}
+</conversation>
+
+<system:instruction>
+Based on the conversation above, generate 1-5 search queries that would be useful for retrieving relevant information from a knowledge base. Focus on the most important topics, questions, or information needs expressed in the conversation.
+
+Make the queries diverse and specific enough to retrieve focused results, but general enough to capture relevant information.
+</system:instruction>`,
+      schema,
+    });
+    
+    return res["queries"];
+  } catch (error) {
+    console.error("Failed to generate search queries:", error);
+    return [];
+  }
+}
+
+/**
+ * Execute multiple search queries in parallel with higher limits
+ * and return combined results
+ */
+async function executeDeepSearchQueries(
+  db: DrizzleDB,
+  userId: string,
+  queries: string[]
+): Promise<string> {
+  // Deep search uses higher limits than real-time search
+  const deepSearchLimit = 20;
+  
+  // Run all search queries in parallel
+  const searchResults = await Promise.all(
+    queries.map(async (query) => {
+      const embedding = await generateQueryEmbedding(query);
+      if (!embedding) return null;
+      
+      return executeSearchWithEmbedding(db, userId, query, embedding, deepSearchLimit);
+    })
+  );
+  
+  // Combine all search results
+  const combinedResult = searchResults
+    .filter(Boolean)
+    .join("\n");
+    
+  return combinedResult || "";
+}
+
+/**
+ * Generate embedding for search query
+ */
+async function generateQueryEmbedding(query: string): Promise<number[] | null> {
```

Comment:
Don't we have an embedding generating method just like this elsewhere already? Let's refactor to make sure we nicely organize and reuse all that's good.